### ISSUE TITLE
counter_culture_fix_counts fails when called on a model backed by a schema-namespaced database table

### ISSUE THREAD

**Issue by smaximov** [magnusvk/counter_culture#184]
Populating counter cache values doesn't seem to work with [schema](https://www.postgresql.org/docs/current/static/ddl-schemas.html)-namespaced database tables. The issue is likely due to `Reconciliation` performs joins using model's table name as the alias which doesn't represent a [valid SQL identifier](https://www.postgresql.org/docs/current/static/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS) in case one is using schematas.

Since your test suite uses SQLite which doesn't support schematas, here is a separate test case for Postgresql:

* Install and configure Postgresql
* Create new database `counter_culture_test`
* Run the following snippet:

    ``` ruby
    begin
      require 'bundler/inline'
    rescue LoadError => e
      $stderr.puts 'Bundler version 1.10 or later is required'
      raise e
    end

    gemfile(true) do
      source 'https://rubygems.org'

      gem 'activerecord', '4.2.5', require: 'active_record'
      gem 'counter_culture', '1.6.2'
      # gem 'counter_culture', github: 'smaximov/counter_culture', ref: '5a4f559'
      gem 'pg', '0.18.4'
      gem 'rspec', '3.6.0', require: 'rspec/autorun'
      gem 'pry'
    end

    require 'logger'

    ActiveRecord::Base.establish_connection(adapter: 'postgresql',
                                            database: 'counter_culture_test')
    ActiveRecord::Base.logger = Logger.new(STDOUT)
    ActiveRecord::Base.raise_in_transactional_callbacks = true

    ActiveRecord::Schema.define do
      execute <<-SQL
        DROP SCHEMA IF EXISTS weblog CASCADE;

        CREATE SCHEMA weblog;

        CREATE TABLE weblog.articles (
          id SERIAL NOT NULL PRIMARY KEY,
          comments_count INTEGER NOT NULL DEFAULT 0
        );

        CREATE TABLE weblog.comments (
          id SERIAL NOT NULL PRIMARY KEY,
          article_id INTEGER NOT NULL REFERENCES weblog.articles(id),
          deleted_at TIMESTAMP WITHOUT TIME ZONE
        );

        INSERT INTO weblog.articles DEFAULT VALUES;

        INSERT INTO weblog.comments(article_id, deleted_at)
             VALUES (currval('weblog.articles_id_seq'), NULL),
                    (currval('weblog.articles_id_seq'), NOW());
      SQL
    end

    module Weblog
      class Article < ActiveRecord::Base
        self.table_name = 'weblog.articles'

        has_many :comments, class_name: '::Weblog::Comment'
      end

      class Comment < ActiveRecord::Base
        COLUMN_NAME = 'comments_count'.freeze

        self.table_name = 'weblog.comments'

        belongs_to :article, class_name: '::Weblog::Article'
        counter_culture :article,
            column_name: ->(comment) { COLUMN_NAME if comment.deleted_at.nil? },
            column_names: {
              ['deleted_at IS NULL'] => COLUMN_NAME
            }
      end
    end

    RSpec.describe Weblog::Comment do
      subject { Weblog::Article.first }

      describe '.counter_culture_fix_counts' do
        it 'handles schema-namespaced tables' do
          expect {
            described_class.counter_culture_fix_counts
          }.to change { subject.reload.comments_count }.from(0).to(1)
        end
      end
    end
    ```

I created a quick fix that makes the above test pass (you need to point `counter_culture` to my fork) in smaximov/counter_culture@5a4f559. I didn't create a PR though, since I'm not familiar with the codebase and pretty sure there are some edge cases that I've missed.

**Comment by magnusvk** [comment]
@smaximov can you see if #188 fixes these issues for you? I need to think about whether I want to add a proper test for this and if so how before I merge into master. But this adds quoting to all the places where we use table names directly (I think) and fixes your ^^ test above.

**Comment by smaximov** [comment]
> can you see if #188 fixes these issues for you?

Yes, it works for me 👍

**Comment by magnusvk** [comment]
@smaximov I couldn't figure out how to write a spec for this with SQLite and I don't want to pull in a Postgres dependency just to test this. :( I merged #188 and released that as 1.8.0, so you can use that version. Fingers crossed this doesn't break in future releases. If it does, please open another issue and we'll figure out a better way to test then.
